### PR TITLE
[FLINK-3195] [examples] Consolidate Java/Scala and batch/streaming examples

### PR DESCRIPTION
**Pull request by StephanEwen**
This pull request consolidates the examples:

  - One root project `flink-examples`
  - Subprojects `flink-examples-batch` and `flink-examples-streaming`
  - Batch examples are put into the distribution under `examples/batch`
  - Streaming examples are now also put into the distribution under `examples/streaming`
  - Java / Scala examples are always in one project. Since Java code never references Scala code, this should be okay, even for Eclipse
  - Removed redundant jars, such as `WordCoundPOJO`, and `EnumTrianglesOpt`. Since the JARs do not add any visible difference, they are not necessary and only confusing.
  - Rename `PageRankBasic` to `PageRank`
  
Open question: Moving the final example jars to `examples/batch` breaks a few web tutorials. Since these tutorials usually mention a Flink version (and this is simple to figure out for users), I think it is okay.

An alternative is to have all examples in the `/examples` folder and changing the names to make them unique, for example `WordCount.jar` and `StreamingWordCount.jar`.

### CODE REVIEW

File: flink-examples/flink-examples-batch/src/main/java/org/apache/flink/examples/java/graph/EnumTrianglesOpt.java [hunk 1]
```diff
@@ -18,25 +18,23 @@
...
 import org.apache.flink.api.common.operators.Order;
-import org.apache.flink.api.java.functions.FunctionAnnotation.ForwardedFields;
-import org.apache.flink.api.java.tuple.Tuple2;
-import org.apache.flink.util.Collector;
 import org.apache.flink.api.java.DataSet;
 import org.apache.flink.api.java.ExecutionEnvironment;
+import org.apache.flink.api.java.functions.FunctionAnnotation.ForwardedFields;
+import org.apache.flink.api.java.tuple.Tuple2;
 import org.apache.flink.examples.java.graph.util.EnumTrianglesData;
-import org.apache.flink.examples.java.graph.util.EnumTrianglesDataTypes.Edge;
```
Comment (fhueske) [inline flink-examples/flink-examples-batch/src/main/java/org/apache/flink/examples/java/graph/EnumTrianglesOpt.java]: `Edge`, `EdgeWithDegrees`, and `Triad` should be added back to the imports to make the code more readable.

File: flink-dist/pom.xml [hunk 2]
```diff
@@ -85,13 +85,7 @@ under the License.
...
-			<artifactId>flink-java-examples</artifactId>
-			<version>${project.version}</version>
-		</dependency>
-
-		<dependency>
-			<groupId>org.apache.flink</groupId>
-			<artifactId>flink-scala-examples</artifactId>
+			<artifactId>flink-examples-batch</artifactId>
 			<version>${project.version}</version>
 		</dependency>
```
Comment (fhueske) [inline flink-dist/pom.xml]: I think we need to add `flink-examples-streaming` as well, no?

File: flink-examples/flink-examples-batch/src/main/java/org/apache/flink/examples/java/ml/util/LinearRegressionData.java [hunk 3]
```diff
@@ -20,8 +20,7 @@
 
 import org.apache.flink.api.java.DataSet;
 import org.apache.flink.api.java.ExecutionEnvironment;
-import org.apache.flink.examples.java.ml.LinearRegression.Data;
```
Comment (fhueske) [inline flink-examples/flink-examples-batch/src/main/java/org/apache/flink/examples/java/ml/util/LinearRegressionData.java]: Re-add these imports?

File: flink-examples/flink-examples-batch/src/main/java/org/apache/flink/examples/java/graph/util/EnumTrianglesData.java [hunk 4]
```diff
@@ -24,7 +24,6 @@
 
 import org.apache.flink.api.java.DataSet;
 import org.apache.flink.api.java.ExecutionEnvironment;
-import org.apache.flink.examples.java.graph.util.EnumTrianglesDataTypes.Edge;
```
Comment (fhueske) [inline flink-examples/flink-examples-batch/src/main/java/org/apache/flink/examples/java/graph/util/EnumTrianglesData.java]: Re-add this import?

**Comment by fhueske** [comment]
I like the restructuring. I am also fine with remove redundant examples. However, I think we should either not rename the triangle enumeration jar to `EnumerateGraphTriangles.jar` or also rename the source file. Otherwise it is not clear which code is contained in the jar. I would also be OK with removing the EnumTriangleOpt example completely.

I think it is OK to move the batch example to a subfolder. They should be rather easy to find.

**Comment by rmetzger** [comment]
I'd like to finish the project restructure and I'll use this PR as a base. I'll also address Fabian's comments.

**Pull request closed without merging**